sources.go

```diff
@@ -113,3 +113,19 @@ func Slice(s []string) *Pipe {
 func Stdin() *Pipe {
 	return NewPipe().WithReader(os.Stdin)
 }
+
+// Prompt presents a message to the user and returns a pipe containing user input or defaultValue provided.
+func Prompt(message, defaultValue string) *Pipe {
+
+	Echo(message).Stdout()
+	p := Stdin().First(1)
+	in, err := p.String()
+	if err != nil {
+		return NewPipe().WithError(err)
+	}
+	in = strings.TrimSpace(in)
```

Comment:
Do we want to silently remove leading and trailing spaces? I can imagine situations where they might be significant.

Reply:
I have done it because using the 
```Stdin().First(1)```
To read user input adds a '\n` character at the end of the input.

```go
//On First
output.WriteString(scanner.Text())
output.WriteRune('\n')
```

A solution would be to utilize a different Stdin read method on ```fmt.Scan``` or ```bufio.Read```

Reply:
I have changed the implementation of user input.

Instead of having ```Stdin().First(1)```, the code now scans the ```Stdin()``` pipe